packages/iTwinUI-react/src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -88,6 +89,68 @@ export const onSingleSelectHandler = <T extends Record<string, unknown>>(
   return newState;
 };
 
+/**
+ * Handles selection when clicked on a row while shift key is pressed.
+ */
+export const onShiftSelectHandler = <T extends Record<string, unknown>>(
+  state: TableState<T>,
+  action: ActionType,
+  instance?: TableInstance<T>,
+  onSelect?: (
+    selectedData: T[] | undefined,
+    tableState?: TableState<T>,
+  ) => void,
+  isRowDisabled?: (rowData: T) => boolean,
+) => {
+  if (instance == null) {
+    return state;
+  }
+
+  let startIndex = instance.flatRows.findIndex(
+    (row) => row.id === state.lastSelectedRow,
+  );
+  let endIndex = instance.flatRows.findIndex((row) => row.id === action.id);
+
+  if (startIndex < 0) {
+    startIndex = 0;
+  }
+  if (endIndex < 0) {
+    endIndex = 0;
+  }
+
+  if (startIndex > endIndex) {
+    const temp = startIndex;
+    startIndex = endIndex;
+    endIndex = temp;
+  }
+
+  const selectedRowIds: Record<string, boolean> = {};
+
+  // 1. Select all rows between start and end
+  instance.flatRows
+    .slice(startIndex, endIndex + 1)
+    .forEach((r) => (selectedRowIds[r.id] = true));
+
+  // 2. Select all children of the last row (endIndex)
+  // Since lastRow's children come after endIndex + 1 (not selected in step 1)
+  const handleRow = (row: Row<T>) => {
+    selectedRowIds[row.id] = true;
+    row.subRows.forEach((r) => handleRow(r));
+  };
+  handleRow(instance.flatRows[endIndex]);
+
+  const newState = {
+    ...state,
+    selectedRowIds: selectedRowIds,
```

Comment:
You can write 
```ts
const newState = {
    ...state,
    selectedRowIds,
}
```
since key and variable names are the same.

Reply:
**[DONE]** Ohh, right, thanks